### PR TITLE
docs: Add the Angular CLI commands

### PR DESCRIPTION
**Pull request by sebawita**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] If there is an issue related with this PR, point it out here.

## What is the new state of the documentation article?
Adds a section on how to use the Angular CLI in a NativeScript project.

### CODE REVIEW

File: docs/tooling/angular-cli.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+---
+title: Angular CLI
+description: Learn how to use the Angular CLI in your NativeScript app.
```
Comment (etabakov) [inline docs/tooling/angular-cli.md]: In terms of SEO, a little bit longer description would be great. Mentioning Schematics as technology, code sharing, would be useful.

**Review by etabakov — APPROVED** [review]

File: docs/tooling/angular-cli.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+environment: angular
+---
+
+# Angular CLI
+
+The **Angular CLI** makes it easy to create an Angular application and generate core building blocks, like **components**, **modules**, **services**, etc.
+
+With the help of the [NativeScript Schematics](https://www.npmjs.com/package/@nativescript/schematics), you can also reap the benefits of the [Angular CLI](https://www.npmjs.com/package/@angular/cli) in your **Angular NativeScript** projects.
+
+The **NativeScript Schematics** extend the **Angular Schematics**, by providing NativeScript-specific functionality and changes to the default **component** and **module** generators.
```
Comment (sis0k0) [inline docs/tooling/angular-cli.md]: I wouldn't mention which of the default schematics are changed and which are kept the same, because this is an implementation detail that may change in the future.

File: docs/tooling/angular-cli.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+{
+  "version": 1,
+  "cli": {
+    "defaultCollection": "@nativescript/schematics"
+  },
+  "projects": {
+    "my-project-name": {
+      "root": "",
+      "sourceRoot": ".",
+       "projectType": "application",
```
Comment (sis0k0) [inline docs/tooling/angular-cli.md]: There's an extra space on this and the next row.

File: docs/tooling/angular-cli.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+       "prefix": "app"
+    }
+  },
+  "defaultProject": "my-project-name"
+}
+```
+
+The **defaultCollection** is where you specify that you want to use the **NativeScript Schematics**, as the default **Schematics collection** for your project.
```
Comment (sis0k0) [inline docs/tooling/angular-cli.md]: I would rephrase that to something like:
```
The **defaultCollection** is where you specify that you want to use the **NativeScript Schematics** when generating building blocks with the Angular CLI.
```

File: docs/tooling/angular-cli.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+or with the `-c` shorthand:
+
+```shorthand
+ng new -c=@nativescript/schematics my-mobile-app
+```
+
+This notifies the Angular CLI that it should use the **ng-new** schematic from the **@nativescript/schematics** npm package. Which in turn creates a new NativeScript Angular project and performs an npm install.
+
+> Note, the NativeScript projects created with the Angular CLI, come with the Angular CLI and NativeScript Schematics **already configured**. There is no need to add the **angular.json** file manually.
```
Comment (sis0k0) [inline docs/tooling/angular-cli.md]: I think that the comma (,) before `come with...` is not necessary.

File: docs/tooling/angular-cli.md [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+You can learn more about code sharing in the [code-sharing section of the docs](../code-sharing/intro).
+
+## Generators
+
+One of the biggest benefits of the Angular CLI is the rapid scaffolding of the application.
+It allows you to generate **components**, **routes**, **services** and **pipes** with a simple `ng generate` (or `ng g` for short) command.
+
+> Feel free to check out the [official Angular CLI documentation for the generate command](https://github.com/angular/angular-cli/wiki/generate), as there is a lot of available options.
+
+In the **NativeScript Schematics** we have extended the schematics for generating **components** and **modules**. The schematics to generate **services**, **pipes**, etc remain the same.
```
Comment (sis0k0) [inline docs/tooling/angular-cli.md]: Same as above. I think that information is not needed here.

File: docs/tooling/angular-cli.md (hunk 3 above)
Comment (sebawita) [reply docs/tooling/angular-cli.md]: I don't understand. What do you mean? What should I change?

File: docs/tooling/angular-cli.md (hunk 2 above)
Comment (sebawita) [reply docs/tooling/angular-cli.md]: good point. fixed

File: docs/tooling/angular-cli.md (hunk 1 above)
Comment (sebawita) [reply docs/tooling/angular-cli.md]: good point. fixed

File: docs/tooling/angular-cli.md (hunk 4 above)
Comment (sebawita) [reply docs/tooling/angular-cli.md]: that makes more sense. fixed

File: docs/tooling/angular-cli.md (hunk 3 above)
Comment (etabakov) [reply docs/tooling/angular-cli.md]: I think @sis0k0 is referring to an extra space in front of "projectType" on line 54

**Review by NickIliev — APPROVED** [review]
